### PR TITLE
[STUD-16] Add CatEntity

### PR DESCRIPTION
**Pull request by MikhailSkrebcov**
[STUD-16] Add CatEntity

trello.com/c/QTAdgHtL

### CODE REVIEW

**Comment by VladSKR** [comment]
Нет описания задачи в коммите

**Comment by VladSKR** [comment]
Нет описания в названии мра

File: src/repository/entity/CatEntity.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package repository.entity;
+
+import java.util.Objects;
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java:4]: Unused import
To clear them automatically, make sure, you use ctrl-alt-L combination

File: src/repository/entity/CatEntity.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+public class CatEntity extends BaseEntity {
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java:7]: Add blank line after class declaration, pls

File: src/repository/entity/CatEntity.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private String Breed;
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: Field names should always start with lowerCase

File: src/repository/entity/CatEntity.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    public String getColor() {
+        return Color;
+    }
+
+    public void setColor(String Color) {
+        this.Color = Color;
+    }
+
+
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java:42]: Unnecessary blank line

File: src/repository/entity/CatEntity.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    @Override
+    public String toString() {
+        return "Breed: " + this.Breed + ", Color: " + this.Color;
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: It is a good practice to add class name in toString method
So toString method would be like, "ClassName(breed = some, color = some)"

**Review by VladSKR — CHANGES_REQUESTED** [review]

File: src/repository/entity/CatEntity.java (hunk 4 above)
Comment (MikhailSkrebcov) [reply src/repository/entity/CatEntity.java:42]: Like!)

Comment (VladSKR) [reply src/repository/entity/CatEntity.java:42]: Well, it still there
You didnt remove it

**Review by NiKLaYDM — COMMENTED** [review]
Бесполезное наследование от BaseEntity

**Review by NiKLaYDM — APPROVED** [review]

File: src/repository/entity/CatEntity.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.time.LocalDate;
+import java.util.UUID;
+
+public class CatEntity extends BaseEntity {
+
+    private String breed;
+    private String color;
+
+    public CatEntity(String breed, String color, UUID id, String name, LocalDate birthDate) {
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: Sort params in right order,please
First - params for superClass, then for this

File: src/repository/entity/CatEntity.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        super(id, name, birthDate);
+        this.breed = breed;
+        this.color = color;
+    }
+
+    public CatEntity() {
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java:12]: Basic constructor should go first, then with params

File: src/repository/entity/CatEntity.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+    }
+
+    public String getBreed() {
+        return this.breed;
+    }
+
+    public void setBreed(String Breed) {
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: lower case param, pls

File: src/repository/entity/CatEntity.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        this.breed = Breed;
+    }
+
+    public String getColor() {
+        return this.color;
+    }
+
+    public void setColor(String Color) {
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: lower case param, pls

File: src/repository/entity/CatEntity.java [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    @Override
+    public LocalDate getBirthDate() {
+        return super.getBirthDate();
+    }
+
+    @Override
+    public void setBirthDate(LocalDate birthDate) {
+        super.setBirthDate(birthDate);
+    }
+
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: You should not override inherited methods, they are the same

File: src/repository/entity/CatEntity.java [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    @Override
+    public String toString() {
+
+        return getClass().getSimpleName() + "(breed = " + this.breed + ", color = " + this.color + ")";
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: id, name, birthDate?

**Review by VladSKR — CHANGES_REQUESTED** [review]

File: src/repository/entity/CatEntity.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+            return false;
+        }
+
+        if (this == object) {
+            return true;
+        }
+
+        CatEntity otherCat = (CatEntity) object;
+
+        return this.breed.equals(otherCat.breed) && this.color.equals(otherCat.color);
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: What would happen if cats have different ids?

File: src/repository/entity/CatEntity.java [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
```
Comment (VladSKR) [inline src/repository/entity/CatEntity.java]: Unnecessary blank line

**Review by VladSKR — CHANGES_REQUESTED** [review]

**Review by VladSKR — APPROVED** [review]

**Pull request merged**